### ISSUE TITLE
acorn-jsx@5 is incompatible with i18n-parser

### ISSUE THREAD

**Issue by connor4312** [i18next/i18next-parser#136]
- Is this a bug or a feature request? *Bug*
- What version are you running (`0.x` or `1.x`)? *beta30*
- Which runner (cli, gulp...)? *gulp*

When trying to parse jsx, a message comes up suggesting you install acorn-jsx in order to do so. However, doing so grabs the latest `5.0.0` which lacks `acorn-jsx/inject`. This causes the message to _still_ show up, because the exception is still captured when trying to load acorn-jsx.

https://github.com/i18next/i18next-parser/blob/a9b843504faf31d0573ba83ee113dc1296afb408/src/lexers/jsx-lexer.js#L38-L46

We should update i18next-parser to be compatible with 5.x, and add acorn-jsx with an appropriate version name as an optional dependency.

```
> require('acorn-jsx');
// all good
> require('acorn-jsx/inject');
Error: Cannot find module 'acorn-jsx/inject'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:50:33)
    at REPLServer.defaultEval (repl.js:240:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:441:10)
```

**Comment by monvillalon** [comment]
I actually ran into this issue and while trying to fix it I ended up upgrading the code for `acorn@6`.
But it comes with a big caveat.  `acorn-es7` is not working yet with `acorn@6`. So this might be a deal breaker for some.

Since then I found out that I really didn't need it and that my problems were related to #140 and misunderstanding how you actually used `acorn-stage3` that also has to be downgraged.

Because of the `acorn-es7` issues unless somebody has a good reason I will not make a PR but anyone can look at the changes at:
https://github.com/monvillalon/i18next-parser/tree/acorn6

**In short in order for everything to work:**

```
yarn install acorn-jsx@4.x acorn-stage3@0.6.0
```

**and your configuration should look like this:**

```
const acornStage3 = require('acorn-stage3/inject');
...
    js: [
      {
        lexer: 'JsxLexer',
        injectors: [acornStage3],
        plugins: {
          stage3: true,
        },
      },
    ],
...
```

**Comment by karellm** [comment]
The main reason I didn't do the upgrade myself is that the plugins we use didn't implement the new plugin structure last time I checked.

I do invite you to use the older version until all plugins are updated. Thanks for the code snippet @monvillalon

**Comment by karellm** [comment]
I'm closing this because in favour of this issue. You can see the [latest update here](https://github.com/i18next/i18next-parser/issues/155#issuecomment-536237811)
